django/db/backends/mysql/validation.py

```diff
@@ -61,4 +61,17 @@ def check_field(self, field, **kwargs):
                     id='mysql.E001',
                 )
             )
+
+        if field.db_index and field_type.lower() in (
+            'tinyblob', 'blob', 'mediumblob', 'longblob', 'tinytext', 'text',
+            'mediumtext', 'longtext', 'json',
+        ):
+            errors.append(
+                checks.Warning(
+                    'A database index on %s is not supported on this database '
```

Comment:
'MySQL does not support a database index on %s columns.' (it's not that the database backend doesn't support it but that the database itself supports it)
hint='An index won't be created. Silence this warning if you don't care about it.'